### PR TITLE
uv5r.py: address 'dropped byte' issue - related to #10505

### PR DESCRIPTION
**Pull request by KC9HI**
# CHIRP PR Checklist

The following must be true before PRs can be merged:

* All tests must be passing.
* Commits should be squashed into logical units.
* Commits should be rebased (or simply rebase-able in the web UI) on current master. Do not put merge commits in a PR.
* Commits in a single PR should be related.
* Major new features or bug fixes should reference a [CHIRP issue](https://chirp.danplanet.com/projects/chirp/issues).
* New drivers should be accompanied by a test image in `tests/images` (except for thin aliases where the driver is sufficiently tested already).
* All files must be GPLv3 licensed or contain no license verbiage. No additional restrictions can be placed on the usage (i.e. such as noncommercial).

Please also follow these guidelines:

* Keep cleanups in separate commits from functional changes.
* Please write a reasonable commit message, especially if making some change that isn't totally obvious (such as adding a new model, adding a feature, etc).
* Do not add new py2-compatibility code (No new uses of `six`, `future`, etc).
* All new drivers should set `NEEDS_COMPAT_SERIAL=False` and use `MemoryMapBytes`.
* New drivers and radio models will affect the Python3 test matrix. You should regenerate this file with `tox -emakesupported` and include it in your commit.

### CODE REVIEW

**Comment by kk7ds** [comment]
The branch name is "1 of2" .. should we go ahead with this or am I waiting for part 2?

File: chirp/drivers/uv5r.py [hunk 1]
```diff
@@ -359,6 +357,19 @@ def _upper_band_from_image(radio):
...
 
 
+def _read_from_data(data, data_start, data_stop):
+    data = data[data_start:data_stop]
+    return data
+
+
+def _get_data_from_image(radio, _data_start, _data_stop):
+    image_data = _read_from_data(
+        radio.get_mmap().get_byte_compatible(),
```
Comment (kk7ds) [inline chirp/drivers/uv5r.py:367]: This `get_byte_compatible()` should be a no-op on anything that is byte-clean. I assume this is just copy pasta and not intentional?

**Comment by KC9HI** [comment]
Yes. Please. I was going to include changes to the baofeng_common.py driver but decided at the last minute to wait to see if there are any any unforeseen problems discovered with this one.

File: chirp/drivers/uv5r.py [hunk 2]
```diff
@@ -1635,34 +1670,20 @@ def apply_offset(setting, obj):
...
         dtmf = RadioSettingGroup("dtmf", "DTMF Settings")
         group.append(dtmf)
 
+        # broadcast FM settings
+
+        # radios with the dropped byte issue also does not expose the broadcast
+        # FM frequency to CHIRP and ignores any frequency provided by CHIRP.
+        # Older radios that do expose the broadcast FM frequency to CHIRP have
+        # a minimum of 3 different know definitions for storing the
+        # frequencies. Sine some frequencies have colisions for some of the
```
Comment (kk7ds) [inline chirp/drivers/uv5r.py:1682]: "since" and "collisions" are misspelled.

**Review by kk7ds — COMMENTED** [review]
If you confirm the `get_byte_compatible()` can be removed, I can snip that out for you if you don't have time to respin.

File: chirp/drivers/uv5r.py (hunk 1 above)
Comment (KC9HI) [reply chirp/drivers/uv5r.py:367]: Thanks. I will look at it after supper.

File: chirp/drivers/uv5r.py (hunk 2 above)
Comment (KC9HI) [reply chirp/drivers/uv5r.py:1682]: Thanks

**Pull request merged**